### PR TITLE
Modal close

### PR DESCRIPTION
**Pull request by DXCanas**
## Summary

Tackling a few bugs:
- https://trello.com/c/cEvFiTVb/368-users-bug-no-user-facing-feedback-of-errors-while-creating-editing-users-e-g-duplicate-user-error
- https://trello.com/c/cEvFiTVb/368-users-bug-no-user-facing-feedback-of-errors-while-creating-editing-users-e-g-duplicate-user-error
- https://trello.com/c/1YRkWjHQ/387-make-view-model-data-access-explicit-rather-than-implicit

### CODE REVIEW

**Comment by indirectlylit** [comment]
I'm seeing some contradictory-looking messaging here:

image

Steps to repro:

* create a user
* attempt to create the user again

Note that the second response returns a `403 Forbidden` with `{"detail":"You do not have permission to perform this action."}`, so this seems in part a server problem - that's the wrong code to send back. (should be maybe `409  Conflict`?)

However we should have slightly better error handling so that we can distinguish the expected error codes (namely these two).

Short-term, and for the purposes of this PR, we should at least make sure the two errors aren't concatenated like that - we should display either what comes back from the server or a message that we know to be accurate, but not both.

**Comment by DXCanas** [comment]
@indirectlylit the second of those isn't returned from the server, it's just a sting in the view. It's just the server error concatenated to that string.

I know it's nasty, I only did it that way because the server error wasn't very informative. It's easy to change, but I included it out of optimism that we'd have better error messages from server in the future. See the comments.

**Comment by indirectlylit** [comment]
yeah we'll need to fix the server response to get better messaging

**Comment by DXCanas** [comment]
Might this be a consequence of add and update user using the same calls? If that's the case, the server might think we're trying to update a user but don't have the correct password.

via CloudMagic Email [https://cloudmagic.com/k/d/mailapp?ct=pa&cv=8.6.36&pv=6.0.1&source=email_footer_2] On Wed, Aug 10, 2016 at 8:26 PM, indirectlylit < notifications@github.com [notifications@github.com] > wrote:
yeah we'll need to fix the server response to get better messaging

File: kolibri/core/assets/src/vue/modal/index.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
 
   <div class="modal-root" v-on:keyup.esc="toggleModal">
-    <div class="modal" v-show="modalstate" transition="modal">
+    <div class="modal" v-if="modalstate" transition="modal">
       <div class="modal-wrapper">
         <div class="modal-container">
           <img @click="toggleModal" class="close-btn" src="./close.svg">
```
Comment (radinamatic) [inline kolibri/core/assets/src/vue/modal/index.vue:7]: ALT attribute? 
If this element is supposed to serve to close the modal, it must be a button - image is not focusable, therefore not navigable by keyboard. See the proper solution below in the `kolibri/plugins/management/assets/src/vue/modal/index.vue` file.

**Comment by indirectlylit** [comment]
This PR is spanning unrelated issues (close buttons and error messaging), and I don't want one to block progress on the other. I'm going to close this - let's attack them in separate PRs.

**Comment by indirectlylit** [comment]
> Might this be a consequence of add and update user using the same calls? If that's the case, the server might think we're trying to update a user but don't have the correct password.

the full message from the server is 

```
ERROR    ValidationError while validating model before checking permissions: ({'__all__': [ValidationError([u'Facility user with this Username and Facility already exists.'])]}, None, None)
[11/Aug/2016 18:32:04] "POST /api/facilityuser/ HTTP/1.1" 403 63
```

**Pull request closed without merging**